Apps/GatewayApi/src/Services/UserProfileService.cs

```diff
@@ -463,5 +465,32 @@ private void QueueEmail(string toEmail, string templateName)
             Dictionary<string, string> keyValues = new() { [EmailTemplateVariable.Host] = hostUrl };
             this.emailQueueService.QueueNewEmail(toEmail, templateName, keyValues);
         }
+
+        private async Task<RequestResult<UserProfileModel>?> ValidateUserProfile(CreateUserRequest createProfileRequest)
+        {
+            // Validate registration age
+            string hdid = createProfileRequest.Profile.HdId;
+            RequestResult<bool> isMinimumAgeResult = await this.ValidateMinimumAge(hdid).ConfigureAwait(true);
```

Comment:
shouldn't this moved into the UserProfileValidator?

Reply:
further discussion below on the UserProfileValidator class comment